nuxeo-retention-web/elements/nuxeo-retention-analytics.js

```diff
@@ -0,0 +1,351 @@
+              <span class="retentionCount">[[activeRetentionCount]]</span>
+            </span>
+            <span class="mainRetentionInfo"> | </span>
+            <span class="mainRetentionInfo"
+              >[[i18n('retentionAnalytics.legalHoldCount')]]
+              <span class="retentionCount">[[legalHoldCount]]</span>
+            </span>
+          </div>
+          <div class="rangeSelector">
+            <div class="dateLabel">
+              <label>[[i18n('retentionAnalytics.from')]]</label>
+            </div>
+            <nuxeo-date-picker value="{{startDate}}"></nuxeo-date-picker>
+            <div class="dateLabel">
+              <label>[[i18n('retentionAnalytics.to')]]</label>
+            </div>
+            <nuxeo-date-picker value="{{endDate}}"></nuxeo-date-picker>
+          </div>
+        </nuxeo-card>
+
+        <div class="flex-layout">
+          <!-- Retention by Document Type -->
+          <nuxeo-repository-data
+            start-date="[[_formatDate(startDate)]]"
+            end-date="[[_extendEndDate(endDate)]]"
+            date-field="dc:created"
+            grouped-by="ecm:primaryType"
+            where='{"ecm:isRecord": "true"}'
+            group-limit="20"
+            data="{{allDocCategoriesCount}}"
+            index="[[index]]"
+          >
+          </nuxeo-repository-data>
+
+          <nuxeo-card heading="[[i18n('retentionAnalytics.retentionByDocType')]]">
+            <chart-pie
+              values="[[_values(allDocCategoriesCount)]]"
+              labels="[[_labels(allDocCategoriesCount)]]"
+              options='{ "legend": { "display": true, "position": "bottom", "labels": { "boxWidth": 12 } }, "animation": false }'
+            >
+            </chart-pie>
+          </nuxeo-card>
+
+          <!-- RETENTION STATUS -->
+          <!-- No Retention -->
+          <nuxeo-repository-data
+            start-date="[[_formatDate(startDate)]]"
+            end-date="[[_extendEndDate(endDate)]]"
+            date-field="dc:created"
+            ecm-mixin-type="HasRelatedText"
+            metrics="cardinality(ecm:uuid)"
+            where='{"ecm:isRecord": "false"}'
+            data="{{noRetention}}"
+            index="[[index]]"
+          >
+          </nuxeo-repository-data>
+
+          <!-- Future Retentions -->
+          <nuxeo-repository-data
+            start-date="[[_formatDate(startDate)]]"
+            end-date="[[_extendEndDate(endDate)]]"
+            date-field="dc:created"
+            grouped-by="ecm:retainUntil"
+            group-limit="10"
+            data="{{futureRetention}}"
+            index="[[index]]"
+          >
+          </nuxeo-repository-data>
+
+          <!-- Expired Retentions -->
+          <nuxeo-repository-data
+            start-date="[[_formatDate(startDate)]]"
+            end-date="[[_extendEndDate(endDate)]]"
+            date-field="dc:created"
+            grouped-by="record:retainUntil"
```

Comment:
Here you can use the same logic of [Here](https://github.com/nuxeo/nuxeo-retention/pull/37/files#r556494736) where you filter and compute the metric at the same time instead of grouping by date and retrieve all informations and compute again the size `expiredRetention.length` in `_valuesRetentionStatus`

Reply:
done